### ISSUE TITLE
download llvm from tarball's rather than shallow clone?

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/tc-build#192]
**Describe the bug**
It looks like downloading llvm is done via git clone.  Might be faster to download the tarball?  I think https://github.com/llvm/llvm-project/tarball/master works.

**To Reproduce**
```
$ ./build-llvm.py --pgo kernel-defconfig --no-ccache
```

**Expected behavior**
Downloads faster.

**Screenshots**

**Environment (please complete the following information):**
  - Command you ran: `./build-llvm.py --pgo kernel-defconfig --no-ccache`
  - Distribution: gLinux (Debian
  - Python version: Python 3.9.12
  - Shell: zsh

**Comment by nathanchance** [comment]
> It looks like downloading llvm is done via git clone.

Right, we do a full LLVM clone by default (not shallow) so that we can jump around in history if we need to as well as reducing bandwidth for updates.

> Might be faster to download the tarball?

I guess it might be; I haven't tried to benchmark that. I suspect that it wouldn't be faster for updates though (over a shallow clone)?

By default, the script downloads its own copy of `llvm-project`. If you already have a copy that you would like to use, you can use the `--llvm-folder` option to tell `build-llvm.py` where it is and it will skip that entire block.
